src/dashboard/src/Layout/NotificationBox.tsx

```diff
@@ -30,29 +31,58 @@ const usePaperStyle = makeStyles(theme => createStyles({
   }
 }))
 
-const NotificationBox: FunctionComponent<BoxProps> = (props) => {
-  const { notifications } = useContext(ConfigContext)
-  const { currentTeamId } = useContext(TeamContext)
-
-  const notification = useMemo(() => {
-    const notification = get(notifications, [currentTeamId])
-    if (notification === undefined) {
-      return get(notifications, ['.default'])
-    }
-    return notification
-  }, [notifications, currentTeamId])
+const ClusterNotificationBox: FunctionComponent<{ cluster: any, setAmlLink: any}> = ({ cluster, setAmlLink }) => {
+  const { data } = useFetch(`/api/clusters/${cluster.id}`, [cluster.id])
+  const notifications = useMemo(() => {
+    if (data === undefined) return []
+    if (!Array.isArray(data.notifications)) return []
+    if (data.amlPortal !== undefined) setAmlLink(data.amlPortal)
+    return data.notifications as string[]
+  }, [data, setAmlLink])
 
   const paperStyle = usePaperStyle()
 
-  if (notification == null || notification === '') return null
+  if (data === undefined) return null
```

Comment:
This judge seems useless since `data === undefined` causes `notification` is `[]`, and the following `map` yields empty children list.